### PR TITLE
feat: add support for postgres

### PR DESCRIPTION
**Pull request by bot190**
Adds a postgresql service to the docker compose. This also updates a few places in the code to generate postgres compatible calls in addition to the sqlite DB calls. 

The hope is that moving to Postgresql will resolve locking issues we're seeing with the sqlite DB.

### CODE REVIEW

File: appdata/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-cssignin.yaml
\ No newline at end of file
+cssignin.yaml
+db
```
Comment (msoucy) [inline appdata/.gitignore]: Would you mind making this `db/` if it's a directory, just for documentation?

File: docker-compose.deploy.yml [hunk 2]
```diff
@@ -27,3 +29,14 @@ services:
       - FLASK_SECRET_KEY
       - CSSIGNIN_CONFIG=/appdata/cssignin.yaml
       - LANG=en_US.UTF-8
+
+  postgres:
+    container_name: chopshop_db
+    image: postgres:16r1
```
Comment (msoucy) [inline docker-compose.deploy.yml]: Should this be `r1` or `rc1`?

File: requirements.txt [hunk 3]
```diff
@@ -22,6 +22,7 @@ python-dateutil~=2.8.2
 pytz~=2023.3
 PyYAML~=6.0
 regex~=2023.5.5
+psycopg2-binary~=2.9.7
```
Comment (msoucy) [inline requirements.txt:25]: What is this used for?

Comment (bot190) [reply requirements.txt:25]: This is the postgresql driver that SQLAlchemy uses.

File: docker-compose.deploy.yml (hunk 2 above)
Comment (bot190) [reply docker-compose.deploy.yml]: They've finally pushed  a postgres:16 tag so using that.

**Review by msoucy — APPROVED** [review]

**Pull request merged**